controllers/configentry_controller_test.go

```diff
@@ -0,0 +1,493 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	logrtest "github.com/go-logr/logr/testing"
+	"github.com/hashicorp/consul-k8s/api/common"
+	"github.com/hashicorp/consul-k8s/api/v1alpha1"
+	capi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type testReconciler interface {
+	Reconcile(req ctrl.Request) (ctrl.Result, error)
+}
+
+func TestConfigEntryControllers_createsConfigEntry(t *testing.T) {
+	t.Parallel()
+	kubeNS := "default"
+
+	cases := []struct {
+		kubeKind       string
+		consulKind     string
+		configEntryCRD common.ConfigEntryResource
+		reconciler     func(client.Client, *capi.Client, logr.Logger) testReconciler
+		compare        func(t *testing.T, consul capi.ConfigEntry)
+	}{
+		{
+			kubeKind:   "ServiceDefaults",
```

Comment:
***whispers*** _can we use constants here which we can keep in the `common` package?_